src/app/map-data/libs/extract-pattern.ts

```diff
@@ -66,7 +91,9 @@ function inferDataType(value: any): string | number | boolean {
   }
 }
 
-export function getHeadersWithTypesCsv(csvString: string): string {
+export function getHeadersWithTypesCsv(
+  csvString: string
+): { csvWithTypes: string; csvEqualToName: string } | string {
```

Comment:
I would say we should maintain types consistent, in this case we can remove return type string/ Pn line 102 you can create an empty object or set types for `csvWithType, csvEqualToName` optional:
```suggestion
): { csvWithTypes?: string; csvEqualToName?: string } {
```

Then return the values null

Reply:
Another option is to return null instead of an empty string